### PR TITLE
Adopt Copier template

### LINKED ISSUE #158: Adopt Copier Template
See https://github.com/bluesky/bluesky/pull/1664 as a template.

Adopt DLS copier template with the following amendments:
- Different license (BSD3 rather than Apache)
- No `.vscode` directory

### PR DESCRIPTION
**Pull request by DiamondJoseph**
Closes #158 

- Additionally ignores .vscode/
- Retains BSD license

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,44 +1,46 @@
...
+    "remoteEnv": {
+        // Allow X11 apps to run inside the container
+        "DISPLAY": "${localEnv:DISPLAY}"
+    },
+    "customizations": {
+        "vscode": {
+            // Set *default* container specific settings.json values on container create.
+            "settings": {
+                "python.defaultInterpreterPath": "/venv/bin/python"
+            },
```
[the remark this answers is not in the document]
Comment (DiamondJoseph) [reply .devcontainer/devcontainer.json:17]: Ticket says NSLS-II don't want .vscode/ directory- are we going to include a .devcontainer that only works with external vscode configuration?

[the remark this answers is not in the document]
Comment (DiamondJoseph) [reply .devcontainer/devcontainer.json:17]: Ah you mean fold these in?
    "python.testing.unittestEnabled": false,
    "python.testing.pytestEnabled": true,
    "editor.formatOnSave": true,
    "editor.codeActionsOnSave": {
        "source.organizeImports": "explicit"
    },
    "[python]": {
        "editor.defaultFormatter": "charliermarsh.ruff",
    },

[the remark this answers is not in the document]
Comment (coretl) [reply .devcontainer/devcontainer.json:17]: yes, into the settings section of `devcontainer.json`

[the remark this answers is not in the document]
Comment (coretl) [reply .devcontainer/devcontainer.json:17]: I think that's what @callumforrester did to bluesky too...

File: README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+[![License](https://img.shields.io/badge/License-BSD_3--Clause-blue.svg)](https://opensource.org/licenses/BSD-3-Clause)
+
+# ophyd_async
+
+Asynchronous Bluesky hardware abstraction code, compatible with control systems like EPICS and Tango
+
+This is where you should write a short paragraph that describes what your module does,
+how it does it, and why people should use it.
+
+<<<<<<< before updating
```
Comment (gilesknap) [inline README.md:14]: Merge error?

File: docs/index.md
Comment (gilesknap) [inline docs/index.md:1]: This is not rendering properly. 

I could not work out why that was the case. 

 also search is not working - is this something up with the Myst config?

**Review by gilesknap — APPROVED** [review]
Looks good. Herculean effort!

File: README.md
[the remark this answers is not in the document]
Comment (gilesknap) [reply README.md:1]: looks good.

**Review by gilesknap — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (DiamondJoseph) [reply README.md]: Ack removed this and the leftover boilerplate

**Comment by gilesknap** [comment]
@DiamondJoseph  🤓 Yay! I am rebased over your branch and my CI is completing. Thanks for this!

I believe this is ready to (rebase and) merge.

**Pull request merged**